pldoc/static/js/colors.js

```diff
@@ -0,0 +1,132 @@
+        init: function() {
+            ColorHandler.setupJSONColors();
+            ColorHandler.listenForClick();
+            ColorHandler.listenForKeypress();
+            ColorHandler.listenForClose();
+        },
+
+        setupJSONColors: function() {
+            $.ajax({
+                async: false,
+                url: '/public/js/colors.json',
+                complete: function(data) {
+                    colors = data.responseJSON;
+                }
+            });
+        },
+
+        listenForClick: function() {
+            var classes;
+
+            $('#colors-preview').on('click', 'button.swatch', function() {
+                $button = $(this);
+                classes = $($button).attr('class').split(/\s+/);
+
+                ColorHandler.showInformationWindow(classes, $button);
+            });
+        },
+
+        listenForKeypress: function() {
+            var key;
+
+            $(document).on('keydown', function(e) {
+                key = e.which;
+
+                if (key === 27) {
+                    if (!$pane.hasClass('is-hidden')) {
+                        ColorHandler.clearValues();
+                        ColorHandler.hideInformationWindow();
+                    }
+                }
+            });
+        },
+
+        listenForClose: function() {
+            $('#colors-preview .info-pane').on('click', '.close-button', function() {
+                ColorHandler.clearValues();
+                ColorHandler.hideInformationWindow();
+            });
+        },
+
+        showInformationWindow: function(classes, button) {
+            var info = false,
+                back = [],
+                backHtml = '';
+
+            $(colors).each(function(i) {
+                if (colors[i].color === classes[1]) {
+                    info = colors[i][classes[2]];
+                }
+            });
+
+            if (info) {
+                $title.text(info[0].title);
+                $desc.text(info[0].description);
+                $ref.text('palette(' + classes[1] + ', ' + classes[2] + ')');
+
+                $(info[0].recs[0].back).each(function(i) {
+                    back.push(info[0].recs[0].back[i]);
+                });
+
+                $('#colors-preview .info-pane').removeClass('is-hidden').focus();
+            }
+
+            if (back.length > 0) {
+                $(back).each(function(i) {
+                    backHtml += '<li>' + back[i] + '</li>';
+                });
+
+                $back.html(backHtml);
+            }
+
+            $hex.text(ColorHandler.rgbaToHex($(button).css('backgroundColor')));
+            $rgb.text($(button).css('backgroundColor'));
+        },
+
+        hideInformationWindow: function() {
+            $pane.addClass('is-hidden');
+            $button.focus();
+        },
+
+        clearValues: function() {
+            $title.text('');
+            $desc.text('');
+            $back.html('');
+        },
+
+        rgbaToHex: function(rgba) {
```

Comment:
I tried including the `AccessibilityColorContrast` from `color-contrast.js` but that file isn't setup to return what I need so it doesn't work, and needs to be duplicated. Unfortunately I don't have time to re-write that file as this needs to be done by like Friday if not sooner.

FWIW, this isn't part of the pattern library that's available for folks to use, so I feel like we have a bit of flexibility here. This is exclusive to the preview site.